cmd/thanos/receive.go

```diff
@@ -895,6 +897,10 @@ func (rc *receiveConfig) registerFlag(cmd extkingpin.FlagClause) {
 		"[EXPERIMENTAL] Enables feature to snapshot in-memory chunks on shutdown for faster restarts.").
 		Default("false").Hidden().BoolVar(&rc.tsdbMemorySnapshotOnShutdown)
 
+	cmd.Flag("tsdb.enable-native-histograms",
```

Comment:
For query and sidecar, do you need to have this flag or native histograms are supported by automatically?

Reply:
This flag is only to enable native histograms for the receiver tsdb: https://github.com/thanos-io/thanos/blob/b8d6b12e24314cf343b56290625ca1c60ef1eae8/cmd/thanos/receive.go#L91

For query and sidecar it's supported by automatically.